src/repr/scalar/datetime.rs

```diff
@@ -1,176 +1,495 @@
-                            match f {
-                                // Do not capture this token because expression
-                                // is going to fail.
-                                Year | Month | Day => None,
-                                // If following part is H:M:S, infer that this
-                                // part is Day. Because this part can use a
-                                // fraction, it should be parsed as PostgreSQL.
-                                _ => {
-                                    // We can capture these annotated tokens
-                                    // because expressions are commutative.
-                                    annotated_parts.push(AnnotatedIntervalPart {
-                                        fmt: TimePartFormat::SQLStandard(f),
-                                        tokens: next_part.clone(),
-                                    });
-                                    Some(TimePartFormat::PostgreSQL(Day))
+        let mut pdt = ParsedDateTime::default();
+
+        let mut value_parts = Vec::new();
+
+        let value_split = value.trim().split_whitespace().collect::<Vec<&str>>();
+        for s in value_split {
+            value_parts.push(tokenize_time_str(s)?);
+        }
+
+        let mut value_parts = value_parts.iter().peekable();
+
+        let mut annotated_parts = Vec::new();
+
+        while let Some(part) = value_parts.next() {
+            let mut fmt = determine_format_w_datetimefield(&part)?;
+            // If you cannot determine the format of this part, try to infer its
+            // format.
+            if fmt.is_none() {
+                fmt = match value_parts.next() {
+                    Some(next_part) => {
+                        match determine_format_w_datetimefield(&next_part)? {
+                            Some(TimePartFormat::SQLStandard(f)) => {
+                                match f {
+                                    // Do not capture this token because expression
+                                    // is going to fail.
+                                    Year | Month | Day => None,
+                                    // If following part is H:M:S, infer that this
+                                    // part is Day. Because this part can use a
+                                    // fraction, it should be parsed as PostgreSQL.
+                                    _ => {
+                                        // We can capture these annotated tokens
+                                        // because expressions are commutative.
+                                        annotated_parts.push(AnnotatedIntervalPart {
+                                            fmt: TimePartFormat::SQLStandard(f),
+                                            tokens: next_part.clone(),
+                                        });
+                                        Some(TimePartFormat::PostgreSQL(Day))
```

Comment:
nit while I'm looking here: standard rust style is to only capitalize the first letter of an acronym: `SqlStandard` and `PostgreSql`.